src/App.tsx

```diff
@@ -14,11 +17,104 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface Type {
+  [key:string]: string,
+}
+
+  type State = {
+    selectGoods: string[],
+    buttonCheck: boolean,
+  };
+
+class App extends React.PureComponent<{}, State> {
+  goodsStorage:Type = { Jam: 'Jam' };
+
+  ending = '';
+
+  state: State = {
+    selectGoods: ['Jam'],
+    buttonCheck: true,
+  };
+
+  isCheck = (good:string) => {
+    return !Object.prototype.hasOwnProperty.call(this.goodsStorage, good);
+  };
+
+  addProduct = (good: string) => {
+    if (!Object.prototype.hasOwnProperty.call(this.goodsStorage, good)) {
+      this.goodsStorage[good] = good;
+    } else {
+      delete this.goodsStorage[good];
+    }
+
+    let selectGood = Object.keys(this.goodsStorage);
+
+    this.state.buttonCheck = !!selectGood.length;
+
+    if (selectGood.length > 1) {
+      selectGood.splice(selectGood.length - 1, 0, ' and ');
+    }
+
+    if (selectGood.length > 3) {
+      selectGood = selectGood.map((ell, _i, arr) => {
+        if (arr.slice(-3).includes(ell)) {
+          return `${ell}`;
+        }
+
+        return `${ell}, `;
+      });
+      console.log(selectGood);
+    }
+
+    this.ending = `${selectGood.length > 1 ? 'are' : 'is'} selected`;
+
+    this.setState({ selectGoods: selectGood });
+  };
+
+  cleanGoods = () => {
+    this.setState({ buttonCheck: false });
+    this.goodsStorage = {};
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1 className="App__title">
+          {this.state.buttonCheck
+            ? `${this.state.selectGoods.join(' ')} ${this.ending}`
+            : 'No goods selected'}
+        </h1>
+        {this.state.buttonCheck
+          && (
+            <button
+              type="button"
+              className="button is-success App__button-remove"
+              onClick={() => this.cleanGoods()}
```

Comment:
if you are not passing any args into the callback function, you can just write {this.cleanGoods}